### ISSUE TITLE
Chapter Move does not load book list Bookstack is accessed remotely (bookstack.mysite.com) but works when accessed locally (192.168.1.5:4444)

### ISSUE THREAD

**Issue by gdmac1** [BookStackApp/BookStack#2151]
**Issue**
Chapter Move page does not load the list of books when bookstack is accessed remotely using bookstack.mysite.com. When I access bookstack from the local network via 192.168.1.5:4444, the feature works as expected.

comment creation and template insertion are also affected (more details below). 

**Steps To Reproduce**
1. Screenshot below of Chapter Move working locally while not working remotely illustrates the issue. similar issue occurs when I try to create comments remotely.
2. Reproduced issue with Chrome and Firefox. Cleared browser data in both cases to rule out browser issue.
3. During testing, i discovered that comment creation & template insertion will not work remotely either. 
3a.Templates will not insert into a page whether you select template, append before or after, or drag and drop.
3b. comments can be typed but when the comment is submitted, the loading symbol shows endlessly as demonstrated in the screenshot and the comment is never added.

**Expected behavior**
See left side of attached image for expected behaviour. Note that bookstack is being accessed locally and works correctly. Right side of screenshot shows the (consistant) behaviour when bookstack is being accessed remotely.

**Screenshots**
![image](https://user-images.githubusercontent.com/7419923/83979277-2b9f1d00-a905-11ea-9a1e-82621c1ffd0a.png)

**Configuration**
 - Exact BookStack Version: BookStack v0.29.3
 - PHP Version:  Unsure, using Linuxserver/bookstack:latest docker image
 - Hosting Method: Docker

**Additional context**
Unsure if this is relevant but I use Hera (https://github.com/aschzero/hera) to make bookstack remotely accessible through cloudflare. Could this setup be causing the issue with these specific features? do these features of bookstack have anything in common that could help me trace a problem with my setup?

Thank you,

**Comment by ssddanbrown** [comment]
Hi @gdmac1,
BookStack does fundamentally not support being hosted via multiple hostnames/ports. You are likely seeing this issue due to BookStack making AJAX calls over the original hostname/port. If you have an `APP_URL` configured, that is the url that will be used.

You'll generally encounter more issues than just AJAX requests as times goes on.

**Comment by gdmac1** [comment]
Hi @ssddanbrown,

Thank you for your reply. I had left APP_URL unset because, when I set APP_URL to bookstack.mysite.com, bookstack would not load correctly. It would break the bookstack UI, as if styling were not loading properly. I had tried other variations with https:// etc, which all caused the same styling issue. 

In the end the problem was a trailing slash. https://bookstack.mysite.com/ was what I needed to set APP_URL to, to get it working.

Thank you for pointing me in the right direction! Everything is now working great.

Thank you
